src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -305,125 +312,122 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 return;
             }
 
-            if (config != null)
+            if (config.GetChildren().Any())
```

Comment:
>config [](http://example.com/codeflow?start=16&length=6)

are we sure we never get `config` with null values? if so can you assert for that?

Reply:
What's the value of the assert when the method has it annotated as "does not accept `null`"?